tests/countries/test_saint_lucia.py

```diff
@@ -0,0 +1,122 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.saint_lucia import SaintLucia, LC, LCA
+from tests.common import CommonCountryTests
+
+
+class TestSaintLucia(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(SaintLucia, years=range(1979, 2050))
+
+    def test_country_aliases(self):
+        self.assertAliases(SaintLucia, LC, LCA)
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertHolidayName(name, (f"{year}-02-22" for year in range(1979, 2050)))
+
+    def test_new_years(self):
+        self.assertHolidayName("New Year's Day", (f"{year}-01-01" for year in range(1979, 2050)))
+
+    def test_first_monday_of_august_holiday(self):
+        self.assertHolidayName("Emancipation Day", (f"{year}-08-01" for year in range(1979, 2050)))
+
+    def test_labour_day(self):
+        self.assertHolidayName("Labour Day", (f"{year}-05-01" for year in range(1979, 2050)))
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        dts = (
+            "1999-04-02",
+            "2000-04-21",
+            "2010-04-02",
+            "2018-03-30",
+            "2019-04-19",
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+        )
+        self.assertHolidayName(name, dts)
+        self.assertHolidayName(name, range(1979, 2050))
+
+    def test_christmas_day(self):
+        self.assertHolidayName("Christmas Day", (f"{year}-12-25" for year in range(1979, 2050)))
+
+    def test_boxing_day(self):
+        self.assertHolidayName("Boxing Day", (f"{year}-12-26" for year in range(1979, 2050)))
+
+    def test_corpus_christi(self):
+        name = "Corpus Christi"
+        dt = (
+            "2018-05-31",
+            "2019-06-20",
+            "2020-06-11",
+            "2021-06-03",
+            "2022-06-16",
+            "2023-06-08",
+            "2024-05-30",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(1979, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        dts = (
+            "1999-04-05",
+            "2000-04-24",
+            "2010-04-05",
+            "2018-04-02",
+            "2019-04-22",
+            "2020-04-13",
+        )
+        self.assertHolidayName(name, dts)
+
```

Comment:
We also need to add tests for Thanksgiving Day and National Day. It will also be useful to extend tests for observed holidays (you can see the example for [Vanuatu](https://github.com/vacanza/holidays/blob/dev/tests/countries/test_vanuatu.py#L43); in this case, you also need to add the `years_non_observed` parameter to the `super().setUpClass` call).
And it would be good to sort the preceding `test_...` methods in holidays chronological order.